### PR TITLE
Cleanup the Druid codebase

### PR DESCRIPTION
**Pull request by fjy**
This PR serves to clean up dead and otherwise unused code in Druid, including features that are just not production ready. Much of the code removed is my code and mistakes I've made in the repo. 

We are now seeking for more contributors to Druid and I removed all code I think will hinder beginners from understanding the code base.

A summary of things removed:
- single segment announcing
- bridge nodes
- configurations to use off heap-based ingestion
- code paths that lead to IndexMaker

### CODE REVIEW

**Comment by gianm** [comment]
@fjy could you summarize the things that you are wanting to remove?

**Comment by himanshug** [comment]
@fjy I suspect if this is going to have merge conflicts with https://github.com/druid-io/druid/pull/1804 which is ready to be merged. 
@xvrl @drcrallen  can you pls check #1804 and merge if it is ok?

**Comment by xvrl** [comment]
agree with @himanshug, we should wait for 1804 so he doesn't have to redo all the work

**Comment by gianm** [comment]
:+1: given that,

- Single segment announcing provided a transition for folks upgrading from 0.6.x, and we've kept it for a couple major versions but should not keep it forever
- Bridge nodes / Offheap ingestion / IndexMaker are not documented and I believe nobody uses them

Also agree we should hold for #1804.

**Comment by gianm** [comment]
failed on `BrokerServerViewTest.testSingleServerAddedRemovedSegment:113->setupZNodeForServer:384 » NodeExists`

**Comment by drcrallen** [comment]
Can we please do PRs that are going to take more than 10 seconds of review out of the druid.io repository?

**Comment by himanshug** [comment]
Agree with @drcrallen about not making branches inside druid-io for these PRs . In fact, I believe, we should "always" create feature/bug-fix branches in our own forks only and send the PR from there. Same will be followed by anyone else making a contribution.

**Comment by himanshug** [comment]
@fjy this PR removes support for multiple properties, does this need any documentation updates?

**Comment by fjy** [comment]
@himanshug good point, will update docs.
@drcrallen noted for future PRs.

**Comment by fjy** [comment]
@himanshug done

**Comment by himanshug** [comment]
:+1:   (after travis)

**Comment by drcrallen** [comment]
@fjy Please help make sure the changes get noted in the 0.9.0 release notes

**Comment by rasahner** [comment]
Our latest test run on the druid.io code failed because the historical node isn't responding.  The log says this:
Exception in thread "main" com.google.inject.CreationException: Guice creation errors:

1) Could not find a suitable constructor in io.druid.server.coordination.ZkCoordinator. Classes must have either one (and only one) constructor annotated with @Inject or a zero-argument constructor that is not private.
  at io.druid.server.coordination.ZkCoordinator.class(ZkCoordinator.java:60)
  at io.druid.cli.CliHistorical$1.configure(CliHistorical.java:78)

ZkCoordinator was changed, and there is not an @Inject annotation on the constructor any more.

**Comment by fjy** [comment]
@rasahner will submit a fix

**Comment by fjy** [comment]
@rasahner I can't reproduce the problem

**Comment by fjy** [comment]
@rasahner ZKCoordinator seems to still have inject...

**Comment by fjy** [comment]
@rasahner Nope I'm wrong, submitting fix

**Comment by fjy** [comment]
@rasahner should be fixed when https://github.com/druid-io/druid/pull/1913 is merged
